tests/test_scrubber.py

```diff
@@ -169,3 +169,22 @@ def test_scrubbing_doesnt_affect_local_vars(sentry_init, capture_events):
     (frame,) = frames
     assert frame["vars"]["password"] == "[Filtered]"
     assert password == "cat123"
+
+
+def test_recursive_EventScrubber(sentry_init, capture_events):
+    sentry_init()
+    events = capture_events()
+    customScrubber = EventScrubber(recursive=True)
+    complex_structure = {
+        "deep": {
+            "deeper": [{"deepest": {"password": "my_darkest_secret"}}],
+        },
+    }
+
+    capture_event({"extra": complex_structure})
+
+    (event,) = events
+    customScrubber.scrub_event(event)
+    assert isinstance(
+        event["extra"]["deep"]["deeper"][0]["deepest"]["password"], AnnotatedValue
+    )
```

Comment:
This is a good test, but the use of the scrubber that we're testing here is a bit disconnected from the whole flow -- you can give the new scrubber to the `init` like a user would `sentry_init(event_scrubber=EventScrubber(recursive=True))` and then it should run on the `event` automatically, so you could get rid of the `.scrub_event` line. (You might have to check for the string `[Filtered]` instead of `AnnotatedValue` if you do it this way since I think the event will already have been serialized at that point.)

Reply:
Ow, I didn't know about the event_scrubber argument. Sure, this saves us some unneeded lines, did it like you suggested, thx for the quick feedback